server/http/restapi/marketplace.go

```diff
@@ -0,0 +1,145 @@
+package restapi
+
+import (
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+type MarketplaceApp struct {
+	Installed bool                     `json:"installed"`
+	Labels    []model.MarketplaceLabel `json:"labels,omitempty"`
+	Manifest  apps.Manifest            `json:"manifest"`
+}
+
+func (a *restapi) handleGetMarketplace(w http.ResponseWriter, req *http.Request, actingUserID string) {
+	filter := req.URL.Query().Get("filter")
+
+	localApps, err := a.getLocalApps()
+	if err != nil {
+		httputils.WriteInternalServerError(w, err)
+		return
+	}
+
+	remoteApps, err := a.getRemoteApps()
+	if err != nil {
+		httputils.WriteInternalServerError(w, err)
+		return
+	}
+
+	apps := mergeApps(remoteApps, localApps)
+
+	// Filter plugins.
+	var result []MarketplaceApp
+	for _, a := range apps {
+		if appMatchesFilter(a.Manifest, filter) {
+			result = append(result, a)
+		}
+	}
+
+	// Sort result alphabetically.
+	sort.SliceStable(result, func(i, j int) bool {
+		return strings.ToLower(result[i].Manifest.DisplayName) <
+			strings.ToLower(result[j].Manifest.DisplayName)
+	})
+
+	httputils.WriteJSON(w, result)
+}
+
+func (a *restapi) getLocalApps() ([]MarketplaceApp, error) {
+	apps, _, err := a.api.Admin.ListApps()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to list local apps")
+	}
+
+	result := make([]MarketplaceApp, len(apps))
+	for i := 0; i < len(apps); i++ {
+		result[i] = MarketplaceApp{
+			Manifest:  *apps[i].Manifest,
+			Installed: true,
+		}
+	}
+
+	return result, nil
+}
+
+func (a *restapi) getRemoteApps() ([]MarketplaceApp, error) {
```

Comment:
Can we add a comment that talks about the difference(s) between local and remote apps?

Reply:
A comment will make much more sense once https://github.com/mattermost/mattermost-plugin-apps/pull/46 is merged.